int32 turns1
int32 steps
int32 turns2
